dbms/src/Flash/Coprocessor/InterpreterUtils.cpp

```diff
@@ -187,13 +188,10 @@ std::tuple<ExpressionActionsPtr, String, ExpressionActionsPtr> buildPushDownFilt
     chain.addStep();
 
     // remove useless tmp column and keep the schema of local streams and remote streams the same.
-    NamesWithAliases project_cols;
     for (const auto & col : analyzer.getCurrentInputColumns())
     {
         chain.getLastStep().required_output.push_back(col.name);
-        project_cols.emplace_back(col.name, col.name);
     }
-    chain.getLastActions()->add(ExpressionAction::project(project_cols));
```

Comment:
I prefer the original code.
Both have the same effect, but the original code is more readable.
Because the original code added the project action explicitly, instead of letting ExpressionActionChain::finalize add it implicitly.

Reply:
The original code get two same projection...

Reply:
ok..

Reply:
Can you add a `RUNTIME_CHECK(!project_after_where.getActions().empty())` after `chain.finalize();chain.clear();`?

Reply:
done